RecoPPS/Local/plugins/DiamondSampicDigiProducer.cc

```diff
@@ -0,0 +1,186 @@
+// -*- C++ -*-
+//
+// Package:    SampicDigi/DiamondSampicDigiProducer
+// Class:      DiamondSampicDigiProducer
+//
+/**\class DiamondSampicDigiProducer DiamondSampicDigiProducer.cc SampicDigi/DiamondSampicDigiProducer/plugins/DiamondSampicDigiProducer.cc
+
+ Description: This plugin takes testbeam data as input and converts them to TotemTimingDigi which could be then passed to reco
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Christopher Misan (krzysztof.misan@cern.ch)
+//         Created:  Sun, 07 Mar 2021 14:42:52 GMT
+//
+//
+
+// system include files
+#include <memory>
+#include <string>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/CTPPSDigi/interface/TotemTimingDigi.h"
+#include "DataFormats/CTPPSDetId/interface/TotemTimingDetId.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/CTPPSDetId/interface/CTPPSDiamondDetId.h"
+
+#include "TFile.h"
+#include "TTree.h"
+#include "TChain.h"
+
+//
+// class declaration
+//
+
+class DiamondSampicDigiProducer : public edm::stream::EDProducer<> {
+public:
+  explicit DiamondSampicDigiProducer(const edm::ParameterSet&);
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+
+  // ----------member data ---------------------------
+  std::vector<std::string> sampicFilesVec;
+  std::unordered_map<unsigned int, std::vector<unsigned int>> detid_vs_chid_;
+  std::unique_ptr<TChain> inputTree;
+
+  //constants
+  const int MAX_SAMPIC_CHANNELS = 32;
```

Comment:
http://cms-sw.github.io/cms_coding_rules.html#2--naming-rules-1
2.17 ... k is preferred for new sets of constants ...
`kMaxSampichannels `

Reply:
(if this goes to /test, the coding rules are not really going to be enforced)

Reply:
naming rules applied